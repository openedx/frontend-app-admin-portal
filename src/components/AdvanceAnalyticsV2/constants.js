// Default group for the analytics. For now, it will be an empty string.
export const DEFAULT_GROUP = '';
