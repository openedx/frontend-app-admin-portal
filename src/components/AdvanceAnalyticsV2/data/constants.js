export const advancedAnalyticsV2QueryKeys = {
  chartCSV: (enterpriseId, chartName) => [...advancedAnalyticsV2QueryKeys.all, 'chartCSV', enterpriseId, chartName],
};

export const CHART_TYPES = {
  // Charts types for Skills tab
  BUBBLE: 'bubble',
  TOP_SKILLS_ENROLLMENT: 'top_skills_enrollment',
  TOP_SKILLS_COMPLETION: 'top_skills_completion',

  // Charts types for Completions tab
  COMPLETIONS_OVER_TIME: 'completions_over_time',
  TOP_COURSES_BY_COMPLETIONS: 'top_courses_by_completions',
  TOP_SUBJECTS_BY_COMPLETIONS: 'top_subjects_by_completions',

  // Charts types for Enrollments tab
  ENROLLMENTS_OVER_TIME: 'enrollments_over_time',
  TOP_COURSES_BY_ENROLLMENTS: 'top_courses_by_enrollments',
  TOP_SUBJECTS_BY_ENROLLMENTS: 'top_subjects_by_enrollments',
  INDIVIDUAL_ENROLLMENTS: 'individual_enrollments',

  // Charts types for Engagements tab
  ENGAGEMENTS_OVER_TIME: 'engagements_over_time',
  TOP_COURSES_BY_ENGAGEMENTS: 'top_courses_by_engagements',
  TOP_SUBJECTS_BY_ENGAGEMENTS: 'top_subjects_by_engagements',
  INDIVIDUAL_ENGAGEMENTS: 'individual_engagements',
};

export const ANALYTICS_TABS = {
  SKILLS: 'skills',
  COMPLETIONS: 'completions',
  ENROLLMENTS: 'enrollments',
  LEADERBOARD: 'leaderboard',
  ENGAGEMENTS: 'engagements',
};
