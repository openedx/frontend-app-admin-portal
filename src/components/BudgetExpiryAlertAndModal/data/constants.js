export const SEEN_ENTERPRISE_EXPIRATION_ALERT_COOKIE_PREFIX = 'seen-enterprise-expiration-alert-';

export const SEEN_ENTERPRISE_EXPIRATION_MODAL_COOKIE_PREFIX = 'seen-enterprise-expiration-modal-';

export const PLAN_EXPIRY_VARIANTS = {
  expired: 'Expired',
  expiring: 'Expiring',
};
