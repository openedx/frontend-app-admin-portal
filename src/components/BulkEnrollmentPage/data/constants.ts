export const SET_SELECTED_ROWS = 'SET SELECTED ROWS';
export const DELETE_ROW = 'DELETE ROW';
export const CLEAR_SELECTION = 'CLEAR SELECTION';
