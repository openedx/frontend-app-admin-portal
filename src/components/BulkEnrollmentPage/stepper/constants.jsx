export const NUM_CONTENT_ITEMS_PER_PAGE = 25;
export const STEPPER_TITLE = 'Learner enrollment';
export const ADD_LEARNERS_TITLE = 'Add learners';
export const REVIEW_TITLE = 'Review selections';
export const NEXT_BUTTON_TEXT = 'Next';
export const PREVIOUS_BUTTON_TEXT = 'Previous';
export const FINAL_BUTTON_TEXT = 'Enroll learners';
export const NEXT_BUTTON_TEST_ID = 'stepper-next';
export const PREV_BUTTON_TEST_ID = 'stepper-prev';
export const FINAL_BUTTON_TEST_ID = 'stepper-final';
export const NOTIFY_CHECKBOX_TEST_ID = 'notify-checkbox';
export const CUSTOMER_SUPPORT_HYPERLINK_TEST_ID = 'customer-support-hyperlink';
export const REVIEW_STEP = 'review';
export const ADD_COURSES_STEP = 'chooseCourses';
export const ADD_COURSES_TITLE = 'Add courses';
export const ALERT_MODAL_TITLE_TEXT = 'An error occurred behind the scenes';
export const ALERT_MODAL_BODY_TEXT = 'Some or all of your enrollments were not successful. Please '
                                     + 'wait a few minutes and try again. If the problem persists, '
                                     + 'please ';
export const SUPPORT_HYPERLINK_TEXT = 'contact enterprise customer support.';
export const SUPPORT_EMAIL_SUBJECT = 'An error occurred during Subscription Enrollment for enterprise ';
export const SUPPORT_EMAIL_BODY = 'The following error occurred when attempting to enroll learners: ';
export const ADD_COURSE_DESCRIPTION = 'By enrolling your learners in courses, you connect your'
                                      + ' learning community with the content that matters most '
                                      + 'to them, and take the guesswork out of getting started '
                                      + 'on the edX platform. To begin, select a course from '
                                      + 'your subscription catalog.';
export const ADD_LEARNERS_DESCRIPTION = 'Select learners with an active or pending subscription '
                                        + 'license to enroll. If you wish to enroll additional '
                                        + 'learners not shown, please first invite them under ';
export const WARNING_ALERT_TITLE_TEXT = 'Too many courses selected';
export const WARNING_ALERT_BODY_TEXT = 'We noticed that you are trying to enroll learners in over 7 '
                                        + 'courses at once - which might be too many for your learners '
                                        + 'to manage. If this is intentional, please continue with this '
                                        + 'enrollment process. If not, please go back and modify your '
                                        + 'course selections.';
