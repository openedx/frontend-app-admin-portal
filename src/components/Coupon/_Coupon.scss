.coupon {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
  transition: opacity 0.2s ease;

  &.dimmed {
    opacity: 0.5;

    &:hover,
    &:focus {
      opacity: 1;
    }
  }

  &:not(.expanded):not(.border-danger) {
    &:hover,
    &:focus {
      border-color: $gray-300 !important; // `!important` needed to override .border Bootstrap class
    }
  }

  &.border-danger {
    .metadata {
      background: $danger-100;

      &:hover,
      &:focus {
        background: $danger-200;
      }
    }
  }

  .metadata {
    transition: background 0.2s ease;
    position: relative;

    &:hover,
    &:focus {
      background: $gray-100;
      cursor: pointer;
    }

    .icons {
      position: absolute;
      top: 0;
      right: 0;

      @include media-breakpoint-up(lg) {
        position: relative;
      }
    }
  }

  &.expanded {
    &:hover,
    &:focus {
      background: inherit;
    }

    .metadata {
      background: $info-900;
      border-radius: 0.1875rem 0.1875rem 0 0;
      color: $white;
    }
  }
}

