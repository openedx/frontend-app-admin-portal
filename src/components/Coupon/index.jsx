import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Icon } from '@edx/paragon';
import { Error, ExpandLess, ExpandMore } from '@edx/paragon/icons';

import CouponDetails from '../../containers/CouponDetails';

import { isTriggerKey, formatTimestamp } from '../../utils';

const triggerKeys = {
  OPEN_DETAILS: [' ', 'Enter'],
  CLOSE_DETAILS: [' ', 'Enter', 'Escape'],
};

class Coupon extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isExpanded: false,
      dimmed: false,
    };

    this.toggleCouponDetails = this.toggleCouponDetails.bind(this);
    this.handleCouponKeyDown = this.handleCouponKeyDown.bind(this);
  }

  componentDidMount() {
    const { isExpanded } = this.props;

    if (isExpanded) {
      this.setState({ // eslint-disable-line react/no-did-mount-set-state
        isExpanded,
      });
    }
  }

  componentDidUpdate(prevProps) {
    const { isExpanded } = this.props;

    if (isExpanded !== prevProps.isExpanded) {
      this.setState({ // eslint-disable-line react/no-did-update-set-state
        isExpanded,
      });
    }
  }

  handleCouponKeyDown(event) {
    const { isExpanded } = this.state;
    if (!isExpanded && isTriggerKey({ triggerKeys, action: 'OPEN_DETAILS', key: event.key })) {
      event.preventDefault();
      this.toggleCouponDetails();
    } else if (isExpanded && isTriggerKey({ triggerKeys, action: 'CLOSE_DETAILS', key: event.key })) {
      event.preventDefault();
      this.toggleCouponDetails();
    }
  }

  /* eslint-disable react/no-unused-class-component-methods */
  /**
   * [tech debt] This class method isn't used by the component itself, but rather by another component
   * that reaches into the class methods of a rendered instance of this component, which should be avoided.
   */
  setCouponOpacity(dimmedStatus) {
    this.setState({
      dimmed: dimmedStatus,
    });
  }
  /* eslint-enable react/no-unused-class-component-methods */

  toggleCouponDetails() {
    this.setState((state) => ({
      isExpanded: !state.isExpanded,
      dimmed: false,
    }));

    if (!this.state.isExpanded) {
      this.props.onExpand();
    } else {
      this.props.onCollapse();
    }
  }

  renderExpandCollapseIcon() {
    const { isExpanded } = this.state;
    const icon = isExpanded ? ExpandLess : ExpandMore;
    const iconColor = isExpanded ? 'text-white' : 'text-primary';
    const screenReaderText = isExpanded ? 'Close' : 'Open';
    return (
      <Icon
        className={classNames('d-inline-block', iconColor)}
        src={icon}
        screenReaderText={`${screenReaderText} coupon details`}
      />
    );
  }

  renderErrorIcon() {
    return (
      <Icon
        className="text-danger mr-2 d-inline-block"
        src={Error}
        screenReaderText="Coupon has an error."
      />
    );
  }

  renderEnrollmentsRedeemed() {
    const {
      data: {
        num_uses: numUses,
        max_uses: maxUses,
      },
    } = this.props;

    const text = maxUses ? `${numUses} of ${maxUses}` : numUses;
    const children = [text];

    if (maxUses) {
      const percentUsed = Math.round((numUses / maxUses) * 100);
      children.push((
        <span key="percent-redemptions-used" className="ml-1">
          {`(${percentUsed}%)`}
        </span>
      ));
    }

    return children;
  }

  render() {
    const { isExpanded, dimmed } = this.state;
    const { data } = this.props;

    return (
      <div
        className={classNames(
          'coupon mb-3 mb-lg-2 rounded border',
          {
            expanded: isExpanded,
            'border-danger': data.errors.length > 0 && !isExpanded,
            dimmed,
          },
        )}
      >
        <div
          className={classNames(
            'metadata',
            'row no-gutters p-2 d-flex align-items-center',
            {
              rounded: !isExpanded,
              'rounded-top': isExpanded,
            },
          )}
          onClick={this.toggleCouponDetails}
          onKeyDown={this.handleCouponKeyDown}
          role="button"
          tabIndex="0"
          aria-expanded={isExpanded}
          aria-controls={`coupon-details-${data.id}`}
        >
          <div className="col-sm-12 col-lg-3 mb-2 mb-lg-0">
            <small className={classNames({ 'text-muted': !isExpanded, 'text-light': isExpanded })}>Coupon Name</small>
            <div>{data.title}</div>
          </div>
          <div className="col-sm-12 col-lg-4 mb-2 mb-lg-0">
            <div className="row no-gutters">
              <div className="col">
                <small className={classNames({ 'text-muted': !isExpanded, 'text-light': isExpanded })}>Valid From</small>
                <div>
                  {formatTimestamp({ timestamp: data.start_date })}
                </div>
              </div>
              <div className="col">
                <small className={classNames({ 'text-muted': !isExpanded, 'text-light': isExpanded })}>Valid To</small>
                <div>
                  {formatTimestamp({ timestamp: data.end_date })}
                </div>
              </div>
            </div>
          </div>
          <div className="col-sm-12 col-lg-4 mb-2 mb-lg-0">
            <div className="row no-gutters">
              <div className="col">
                <small className={classNames({ 'text-muted': !isExpanded, 'text-light': isExpanded })}>Assignments Remaining</small>
                <div>{data.num_unassigned}</div>
              </div>
              <div className="col">
                <small className={classNames({ 'text-muted': !isExpanded, 'text-light': isExpanded })}>Enrollments Redeemed</small>
                <div>
                  {this.renderEnrollmentsRedeemed()}
                </div>
              </div>
            </div>
          </div>
          <div className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0">
            {data.errors.length > 0 && !isExpanded && this.renderErrorIcon()}
            {this.renderExpandCollapseIcon()}
          </div>
        </div>
        <CouponDetails
          isExpanded={isExpanded}
          couponData={data}
        />
      </div>
    );
  }
}

Coupon.defaultProps = {
  isExpanded: false,
  onExpand: () => {},
  onCollapse: () => {},
};

Coupon.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    start_date: PropTypes.string.isRequired,
    end_date: PropTypes.string.isRequired,
    errors: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
    num_unassigned: PropTypes.number.isRequired,
    num_uses: PropTypes.number.isRequired,
    max_uses: PropTypes.number,
  }).isRequired,
  isExpanded: PropTypes.bool,
  onExpand: PropTypes.func,
  onCollapse: PropTypes.func,
};

export default Coupon;
