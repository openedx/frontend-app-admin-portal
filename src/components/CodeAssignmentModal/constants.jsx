import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
import { MODAL_TYPES } from '../EmailTemplateForm/constants';
import { getTemplateEmailFields } from '../EmailTemplateForm';
import CheckboxWithTooltip from '../ReduxFormCheckbox/CheckboxWithTooltip';

import messages from './messages';

export const ASSIGNMENT_ERROR_TITLES = {
  [MODAL_TYPES.assign]: 'Unable to assign codes',
  [MODAL_TYPES.save]: 'Unable to save template',
};
export const EMAIL_TEMPLATE_NUDGE_EMAIL_ID = 'enable-nudge-emails';

export const getAssignmentModalFields = formatMessage => {
  const emailTemplateFields = getTemplateEmailFields(formatMessage);
  return {
    ...emailTemplateFields,
    [EMAIL_TEMPLATE_NUDGE_EMAIL_ID]: {
      name: EMAIL_TEMPLATE_NUDGE_EMAIL_ID,
      id: EMAIL_TEMPLATE_NUDGE_EMAIL_ID,
      component: CheckboxWithTooltip,
      className: 'auto-reminder-wrapper',
      icon: faInfoCircle,
      altText: formatMessage(messages.modalAltText),
      tooltipText: formatMessage(messages.modalTooltipText),
      label: formatMessage(messages.modalFieldLabel),
      defaultChecked: true,
    },
  };
};

export const NOTIFY_LEARNERS_CHECKBOX_TEST_ID = 'notify-learners-checkbox';
export const SUBMIT_BUTTON_TEST_ID = 'submit-button';
