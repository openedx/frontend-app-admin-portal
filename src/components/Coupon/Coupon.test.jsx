import React from 'react';
import { Provider } from 'react-redux';
import {
  fireEvent, render, screen, waitFor,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import { MULTI_USE } from '../../data/constants/coupons';

import Coupon from './index';

const enterpriseId = 'test-enterprise';
const mockStore = configureMockStore([thunk]);
const store = mockStore({
  portalConfiguration: {
    enterpriseId,
  },
  table: {
    'coupon-details': {
      loading: false,
      results: [],
    },
  },
  csv: {
    'coupon-details': {},
  },
  coupons: {
    couponOverviewLoading: false,
    couponOverviewError: null,
  },
});

const initialCouponData = {
  id: 1,
  title: 'Test Coupon',
  start_date: '2018-06-01T12:00:00Z',
  end_date: '2019-02-01T12:00:00Z',
  errors: [],
  num_unassigned: 0,
  num_uses: 0,
  max_uses: 1,
  usage_limitation: MULTI_USE,
};

const CouponWrapper = props => (
  <MemoryRouter>
    <Provider store={store}>
      <IntlProvider locale="en">
        <Coupon
          data={initialCouponData}
          {...props}
        />
      </IntlProvider>
    </Provider>
  </MemoryRouter>
);

describe('<Coupon />', () => {
  describe('renders correctly', () => {
    it('with max uses', () => {
      const coupon = renderer.create((<CouponWrapper />)).toJSON();
      expect(coupon).toMatchSnapshot();
    });

    it('without max uses', () => {
      const coupon = renderer
        .create(
          <CouponWrapper
            data={{
              ...initialCouponData,
              max_uses: null,
            }}
          />,
        )
        .toJSON();
      expect(coupon).toMatchSnapshot();
    });

    it('with error state', () => {
      const coupon = renderer
        .create(
          <CouponWrapper
            data={{
              ...initialCouponData,
              errors: [{ code: 'test-code', user_email: 'test@example.com' }],
            }}
          />,
        )
        .toJSON();
      expect(coupon).toMatchSnapshot();
    });
  });

  describe('expands and collapses correctly', () => {
    it('expands on click of collapsed coupon', async () => {
      const mockOnExpand = jest.fn();
      render(<CouponWrapper onExpand={mockOnExpand} />);
      await waitFor(() => {
        userEvent.click(screen.getByText(initialCouponData.title));
        expect(mockOnExpand).toBeCalledTimes(1);
      });
    });

    it('collapses on click of expanded coupon', async () => {
      const mockOnCollapse = jest.fn();
      render(<CouponWrapper onCollapse={mockOnCollapse} />);
      await waitFor(() => {
        userEvent.click(screen.getByText(initialCouponData.title));
        expect(mockOnCollapse).toBeCalledTimes(1);
      });
    });

    it('expands on keydown of expanded coupon', () => {
      const mockOnExpand = jest.fn();
      render(<CouponWrapper onExpand={mockOnExpand} />);
      fireEvent.keyDown(screen.getByRole('button'), { key: 'Enter', code: 'Enter', charCode: 13 });
      expect(mockOnExpand).toBeCalledTimes(1);
    });

    it('collapses on keydown of expanded coupon', () => {
      const mockOnCollapse = jest.fn();
      render(<CouponWrapper onCollapse={mockOnCollapse} />);

      fireEvent.keyDown(screen.getByRole('button'), { key: 'Enter', code: 'Enter', charCode: 13 });
      fireEvent.keyDown(screen.getAllByRole('button')[0], { key: 'Enter', code: 'Enter', charCode: 13 });
      expect(mockOnCollapse).toBeCalledTimes(1);
    });

    it('does not handle unknown keydown event', () => {
      const mockOnExpandOrOnCollapse = jest.fn();

      render(
        <CouponWrapper
          onExpand={mockOnExpandOrOnCollapse}
          onCollapse={mockOnExpandOrOnCollapse}
        />,
      );

      fireEvent.keyDown(screen.getByRole('button'), { key: 'A', code: 'KeyA' });
      expect(mockOnExpandOrOnCollapse).toBeCalledTimes(0);
    });
  });
});
