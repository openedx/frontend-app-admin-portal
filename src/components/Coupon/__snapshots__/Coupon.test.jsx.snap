// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Coupon /> renders correctly with error state 1`] = `
<div
  className="coupon mb-3 mb-lg-2 rounded border border-danger"
>
  <div
    aria-controls="coupon-details-1"
    aria-expanded={false}
    className="metadata row no-gutters p-2 d-flex align-items-center rounded"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
    >
      <small
        className="text-muted"
      >
        Coupon Name
      </small>
      <div>
        Test Coupon
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid From
          </small>
          <div>
            June 1, 2018
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid To
          </small>
          <div>
            February 1, 2019
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Assignments Remaining
          </small>
          <div>
            0
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Enrollments Redeemed
          </small>
          <div>
            0 of 1
            <span
              className="ml-1"
            >
              (0%)
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
    >
      <span
        className="pgn__icon text-danger mr-2 d-inline-block"
      >
        <svg
          aria-hidden={true}
          fill="none"
          focusable={false}
          height={24}
          role="img"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2Zm1 15h-2v-2h2v2Zm0-4h-2V7h2v6Z"
            fill="currentColor"
          />
        </svg>
        <span
          className="sr-only"
        >
          Coupon has an error.
        </span>
      </span>
      <span
        className="pgn__icon d-inline-block text-primary"
      >
        <svg
          aria-hidden={true}
          fill="none"
          focusable={false}
          height={24}
          role="img"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41Z"
            fill="currentColor"
          />
        </svg>
        <span
          className="sr-only"
        >
          Open coupon details
        </span>
      </span>
    </div>
  </div>
  <div
    className="coupon-details row no-gutters px-2 my-3 d-none"
    id="coupon-details-1"
  >
    <div
      className="col"
    />
  </div>
</div>
`;

exports[`<Coupon /> renders correctly with max uses 1`] = `
<div
  className="coupon mb-3 mb-lg-2 rounded border"
>
  <div
    aria-controls="coupon-details-1"
    aria-expanded={false}
    className="metadata row no-gutters p-2 d-flex align-items-center rounded"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
    >
      <small
        className="text-muted"
      >
        Coupon Name
      </small>
      <div>
        Test Coupon
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid From
          </small>
          <div>
            June 1, 2018
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid To
          </small>
          <div>
            February 1, 2019
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Assignments Remaining
          </small>
          <div>
            0
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Enrollments Redeemed
          </small>
          <div>
            0 of 1
            <span
              className="ml-1"
            >
              (0%)
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
    >
      <span
        className="pgn__icon d-inline-block text-primary"
      >
        <svg
          aria-hidden={true}
          fill="none"
          focusable={false}
          height={24}
          role="img"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41Z"
            fill="currentColor"
          />
        </svg>
        <span
          className="sr-only"
        >
          Open coupon details
        </span>
      </span>
    </div>
  </div>
  <div
    className="coupon-details row no-gutters px-2 my-3 d-none"
    id="coupon-details-1"
  >
    <div
      className="col"
    />
  </div>
</div>
`;

exports[`<Coupon /> renders correctly without max uses 1`] = `
<div
  className="coupon mb-3 mb-lg-2 rounded border"
>
  <div
    aria-controls="coupon-details-1"
    aria-expanded={false}
    className="metadata row no-gutters p-2 d-flex align-items-center rounded"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
    >
      <small
        className="text-muted"
      >
        Coupon Name
      </small>
      <div>
        Test Coupon
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid From
          </small>
          <div>
            June 1, 2018
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Valid To
          </small>
          <div>
            February 1, 2019
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Assignments Remaining
          </small>
          <div>
            0
          </div>
        </div>
        <div
          className="col"
        >
          <small
            className="text-muted"
          >
            Enrollments Redeemed
          </small>
          <div>
            0
          </div>
        </div>
      </div>
    </div>
    <div
      className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
    >
      <span
        className="pgn__icon d-inline-block text-primary"
      >
        <svg
          aria-hidden={true}
          fill="none"
          focusable={false}
          height={24}
          role="img"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41Z"
            fill="currentColor"
          />
        </svg>
        <span
          className="sr-only"
        >
          Open coupon details
        </span>
      </span>
    </div>
  </div>
  <div
    className="coupon-details row no-gutters px-2 my-3 d-none"
    id="coupon-details-1"
  >
    <div
      className="col"
    />
  </div>
</div>
`;
